x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java

```diff
@@ -105,6 +110,10 @@ protected void onStartJob(long now) {
 
     @Override
     protected SearchRequest buildSearchRequest() {
+        // if run for the 1st time, create checkpoint
+        if (getPosition() == null) {
+            createCheckpoint();
+        }
```

Comment:
Do you agree that after some refactoring, that this belongs in `onStart` or something instead of here? Since `onStart` (or whatever the method will be called) will only be called when there is the transition from STARTED -> INDEXING.

Reply:
absolutely, now that onStart is after the state change ,I will refactor this and the finishing part after the other PR is in.